pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -377,7 +377,129 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
     ***************************************************************************/
 
     /// @inheritdoc ICompositeLiquidityRouter
-    function removeLiquidityProportionalFromNestedPools(
+    function addLiquidityUnbalancedNestedPool(
+        address parentPool,
+        address[] memory tokensIn,
+        uint256[] memory exactAmountsIn,
+        uint256 minBptAmountOut,
+        bytes memory userData
+    ) external returns (uint256) {
+        return
+            abi.decode(
+                _vault.unlock(
+                    abi.encodeWithSelector(
+                        CompositeLiquidityRouter.addLiquidityUnbalancedNestedPoolHook.selector,
+                        AddLiquidityHookParams({
+                            pool: parentPool,
+                            sender: msg.sender,
+                            maxAmountsIn: exactAmountsIn,
+                            minBptAmountOut: minBptAmountOut,
+                            kind: AddLiquidityKind.UNBALANCED,
+                            wethIsEth: false,
```

Comment:
We might need to address this one, as there could be wrapped eth in one of the pools.
Can be done in a follow-up PR.